protein_analysis_tool.py

```diff
@@ -0,0 +1,491 @@
+    - get_amino_acid_sum procedure returns list of dictionaries
+    - codon_optimization procedure returns list of strings
+    - length procedure returns list of int values
+    - brutto_count procedure returns list of dictionaries with counts of atoms in the sequence
+    """
+    amino_acid_seqs = name_transform(args, letter_format)
+    procedures = {
+        "molecular_weight": molecular_weight,
+        "one_letter_to_three": one_letter_to_three,
+        "get_amino_acid_sum": get_amino_acid_sum,
+        "codon_optimization": codon_optimization,
+        "length": length,
+        "brutto_count": brutto_count,
+    }
+    if procedure not in procedures.keys():
+        raise ValueError("Requested procedure is not defined")
+    elif procedure == "codon_optimization":
+        return procedures.get(procedure)(amino_acid_seqs, cell_type)
+    else:
+        return procedures.get(procedure)(amino_acid_seqs)
+
+
+def molecular_weight(amino_acid_seqs: list) -> list:
+    """
+    Calculates predicated molecular weight of aa sequences.
+
+     Arguments:
+    - amino_acid_seqs (list): list of string with the protein sequences
+
+    Return:
+    - List of of floats corresponding to the molecular weight in kDa
+    """
+    molecular_weights = []
+    for seq in amino_acid_seqs:
+        total_weight = 0
+        for aa in seq:
+            aa = aa.upper()
+            total_weight += amino_weights[aa]
+        molecular_weights.append(round(total_weight / 1000, 2))
+    return molecular_weights
+
+
+def one_letter_to_three(amino_acid_seqs: list) -> list:
+    """
+    Translates one-letter coded amino acid sequences to three-letter coded
+    Arguments:
+    - amino_acid_seqs (list): list of string with the protein sequences
+
+    Return:
+    - List of of strings with three-letter coded sequences
+    """
+    three_letters_seqs = []
+    for seq in amino_acid_seqs:
+        three_letters_seq = []
+        for aa in seq:
+            aa = aa.upper()
+            three_letters_seq.append(amino_short_names_dic[aa])
+        three_letters_seqs.append("".join(three_letters_seq))
+    return three_letters_seqs
+
+
+def get_amino_acid_sum(protein_sequences: list) -> list:
+    """
+    Counts the amount of each amino acid in the injected protein sequences
+
+    Arguments:
+    - protein_sequences (list): list of injected protein sequence
+
+    Return:
+    - List of dictionary with amino acid amount"""
+    result = []
+    for protein_sequence in range(len(protein_sequences)):
+        amino_acid_count = {
+            "A": 0,
+            "C": 0,
+            "D": 0,
+            "E": 0,
+            "F": 0,
+            "G": 0,
+            "H": 0,
+            "I": 0,
+            "K": 0,
+            "L": 0,
+            "M": 0,
+            "N": 0,
+            "P": 0,
+            "Q": 0,
+            "R": 0,
+            "S": 0,
+            "T": 0,
+            "V": 0,
+            "W": 0,
+            "Y": 0,
+        }
```

Comment:
Лучше было погуглить и создать словарь с 0 на лету. Пример:
```suggestion
        amino_acid_count = dict([(key, 0) for key in amino_short_names_dic.keys()])
```